ccmbioinfo/lfs
# CARDS-1375: Add an option to block advancing in a form unless all required fields have been filled out

**Overview of changes visible to the user:**
* A new questionnaire property labeled **Require completion** has been added and can be updated in the questionnaire designer
* The PREMs surveys (CPESIC, OAIP, OED, Rehab) have been updated to have this property set to true
* When a form requires completion & the user saves or clicks next in the pagination, the page scrolls to the 1st incomplete question on the screen which gets outlined in red

**Testing**
* Check that nothing changed when filling out a form that does NOT require completion, both as a normal user and as a patient
* Fill in forms with required completion and check that the behavior matches the description above
* Fill in the PREMs surveys as a patient and check that you can't advance to the next page not to the next survey if all the questions on the current screen aren't complete

**Regression testing**
Test that the code refactoring didn't affect the following:
* The color of the answer instructions (e.g. "This question is mandatory") correctly reflects the presence of the answer in all questions including QuestionMatrix
* Incomplete questions and sections appear in view mode, while non-mandatory questions without an answer are not displayed in view mode
